momentGW/pbc/tda.py

```diff
@@ -242,12 +307,61 @@ def build_se_moments(self, moments_dd):
                         Lp = self.integrals.Lpx[kp, kx][:, :, x]
                         subscript = f"P{pchar},Q{qchar},PQ->{pqchar}"
                         eta[kp, q][x, n] += util.einsum(subscript, Lp, Lp.conj(), eta_aux)
+                        if self.fc and q == 0:
+                            original = eta[kp, q][x, n]
+
+                            eta[kp, q][x, n] += (2 / np.pi) * (q0) * eta_head[kp, n] * original
+
+                            wing_tmp = util.einsum("Pp,P->p", Lp, eta_wings[kp, n])
+                            wing_tmp = wing_tmp.real * 2
+                            wing_tmp *= -(np.sqrt(cell_vol / (4 * (np.pi**3))) * q0**2)
+
+                            eta[kp, q][x, n] += util.einsum("p,pq->pq", wing_tmp, original)
```

Comment:
Same comment as the `moments_dd`. Is it possible to have this function be unchanged, but before we do the convolution, there is something like:
```
if self.fc:
    eta = self._add_eta_fc_correction(moments_dd)

# Construct the self-energy moments
moments_occ, moments_vir = self.convolve(eta)
```

Reply:
Have been able to do this and split `moments_dd` as well.